node/src/validator/mod.rs

```diff
@@ -133,6 +131,9 @@ impl<N: Network, C: ConsensusStorage<N>> Validator<N, C> {
         )
         .await?;
 
+        // Initialize the sync module.
+        let sync = BlockSync::new(BlockSyncMode::Gateway, ledger_service, router.tcp().clone());
```

Comment:
@raychu86 can you review to confirm this reordering is safe? IIRC this has initialization implications to the node's state.

Reply:
This particular reordering should be safe - `BlockSync::new` now requires an objects from `Router`, which is why it is initialized after Router. `Router` initialization is not affected and does not affect node state.